File: Java/Medium/Accepted/super-pow_2.java
class Solution {
    public int superPow(int a, int[] b) {
        
        int res = 1;
        for (int i : b) {
            res = pow(res, 10) * pow(a, i) % 1337;
        }
        return res;
    }
    private int pow(int a, int k) {
        a %= 1337;
        int res = 1;
        for (int i = 0; i < k; i++) {
            res = res * a % 1337;
        }
        return res;
    }
}

File: Rust/Medium/WrongAnswer/super-pow_2.rs
impl Solution {
    pub fn super_pow(a: i32, b: Vec<i32>) -> i32 {
        

        let mut v = 1;
        let mut a = a;
        for i in b.iter().rev() {
            v = v * (a.pow(*i as u32) % 1337) % 1337;
            a = a.pow(10) % 1337;
        }
        v
    }
}